dev/spotbugs-exclude.xml

```diff
@@ -0,0 +1,91 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one
+  ~ or more contributor license agreements.  See the NOTICE file
+  ~ distributed with this work for additional information
+  ~ regarding copyright ownership.  The ASF licenses this file
+  ~ to you under the Apache License, Version 2.0 (the
+  ~ "License"); you may not use this file except in compliance
+  ~ with the License.  You may obtain a copy of the License at
+  ~
+  ~   http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+<FindBugsFilter>
```

Comment:
@Fokko please add a comment explaining that all default rules are excluded to enable them one-by-one in the future. Also, did you check that there is no way to disable all rules at once?

Reply:
Yes, I've added a comment to the XML. I think it makes more sense to have all the rules excluded individually, so we can enable them one by one. In this process, we must be sure that PR's are properly rebased against latest master in order to make sure that it passes the Spotbugs checks. Otherwise, I'm fine as well with fixing these violations. WDYT?

Reply:
So the approach with all-exclusions requires proper rebase, or vice versa?

Also, I think it's better to put this file in `codestyle/` directory than `dev/`. `dev/` is more for files to be consumed by people, and `codestyle/` for files to be consumed by the build.

Reply:
Good point, I've moved the file to `codestyle/`.

An all-exclusions would not work at all in my opinion, because that would require to fix all the violations at once, which is virtually undoable.

In the situation where we enable one of the rules by a PR down the road. And let's say there is a PR which branched from a commit where this rule wasn't enabled yet (behind master), this will break the CI on the master branch with a spotbugs voilation. This isn't a huge issue since it will be a small change to get it fixed again.